File: apps/desktop/src/shared/strip-ansi.ts
// CSI / OSC / single-char ANSI escape sequences from terminal output.
const ANSI_ESCAPE_PATTERN =
  /(?:\u001B\][^\u0007]*(?:\u0007|\u001B\u005C|\u009C))|(?:\u001B[@-Z\\-_])|(?:\u001B\[[0-?]*[ -/]*[@-~])/g;

export function stripAnsi(value: string): string {
  return value.replace(ANSI_ESCAPE_PATTERN, "");
}

File: apps/mobile/lib/core/utils/strip_ansi.dart
// CSI / OSC / single-char ANSI escape sequences from terminal output.
final _ansiEscapePattern = RegExp(
  r'(?:\u001B\][^\u0007]*(?:\u0007|\u001B\\|\u009C))|(?:\u001B[@-Z\\-_])|(?:\u001B\[[0-?]*[ -/]*[@-~])',
);

String stripAnsi(String value) {
  return value.replaceAll(_ansiEscapePattern, '');
}
